### PR TITLE
chore: remove deprecated Dataplane properties

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## What this PR changes/adds

Removes the deprecated `turnCount` and `allowedDestTypes` from the DataPlaneInstance

## Why it does that

deprecated since 0.6 and 0.7 respectively

## Further notes

_List other areas of code that have changed but are not necessarily linked to the main feature. This could be method
signature changes, package declarations, bugs that were encountered and were fixed inline, etc._

## Linked Issue(s)

Closes # <-- _insert Issue number if one exists_

_Please be sure to take a look at the [contributing guidelines](https://github.com/eclipse-edc/.github/blob/main/CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](https://github.com/eclipse-edc/.github/blob/main/contributing/pr_etiquette.md)._

### CODE REVIEW

File: extensions/control-plane/transfer/transfer-data-plane-signaling/src/test/java/org/eclipse/edc/connector/controlplane/transfer/dataplane/flow/DataPlaneSignalingFlowControllerTest.java [hunk 1]
```diff
@@ -66,6 +66,36 @@ public class DataPlaneSignalingFlowControllerTest {
             () -> URI.create("http://localhost"), selectorService, propertiesProvider, dataPlaneClientFactory,
             "random", transferTypeParser);
 
+    @NotNull
+    private DataPlaneInstance.Builder dataPlaneInstanceBuilder() {
```
Comment (ndr-brt) [inline extensions/control-plane/transfer/transfer-data-plane-signaling/src/test/java/org/eclipse/edc/connector/controlplane/transfer/dataplane/flow/DataPlaneSignalingFlowControllerTest.java:70]: I know that this is likely some automated plugin, but moving private methods above the nested classes that contains the tests causes readability issues.
Automated conventions could be nice, but I would say that in cases like this they could be avoided.

generally speaking, I prefer an approach where "most important stuff" stays at the top, while "least important thing" stays at the bottom.

Comment (paullatzelsperger) [reply extensions/control-plane/transfer/transfer-data-plane-signaling/src/test/java/org/eclipse/edc/connector/controlplane/transfer/dataplane/flow/DataPlaneSignalingFlowControllerTest.java:70]: I agree, but the problem with that while it is possible to sort "nested class before private method", there is no code style rule in IntelliJ (to my knowledge) that lets you specify that by class annotation `@Nested`. 

However, it is possible to arrange "nested classes - private/protected method - static nested classes".

**Review by ndr-brt — COMMENTED** [review]
another thing that I forgot, the change needs to be reflected on `DataPlaneInstanceSchema` that will make this become a breaking change on management api v3:
https://eclipse-edc.github.io/Connector/openapi/management-api/#/Dataplane%20Selector%20V3/getAllDataPlaneInstancesV3

File: extensions/control-plane/transfer/transfer-data-plane-signaling/src/test/java/org/eclipse/edc/connector/controlplane/transfer/dataplane/flow/DataPlaneSignalingFlowControllerTest.java (hunk 1 above)
Comment (ndr-brt) [reply extensions/control-plane/transfer/transfer-data-plane-signaling/src/test/java/org/eclipse/edc/connector/controlplane/transfer/dataplane/flow/DataPlaneSignalingFlowControllerTest.java]: could it be applied on the other test classes that are involved in the PR?

**Comment by paullatzelsperger** [comment]
> another thing that I forgot, the change needs to be reflected on `DataPlaneInstanceSchema` that will make this become a breaking change on management api v3: https://eclipse-edc.github.io/Connector/openapi/management-api/#/Dataplane%20Selector%20V3/getAllDataPlaneInstancesV3

ok, to do this properly, we'll have to:
- introduce a `v4-alpha` version of the Management API
- not serialize `allowedDestTypes` and `turnCount` through it
- once `v4` is stable, deprecate `v3`
- when `v3` is finally removed, remove the properties from the `DataPlaneInstance`

All in all, this PR should be closed :)

**Pull request closed without merging**